### PR TITLE
KZOO-229: adding appex client to core

### PR DESCRIPTION
**Pull request by icehess**

### CODE REVIEW

**Comment by icehess** [comment]
Circular Dependent PR:
* https://github.com/2600hz/kazoo-crossbar/pull/268

File: kazoo_appex/src/appex_bluepak.erl [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+      ,{'attribute',2,'behaviour','application'}
+      ,{'attribute',3,'export',[{'start',2},{'stop',1}]}
+      ,{'function',4,'start',2
+       ,[{'clause',4, [{'var',4,'_'},{'var',4,'_'}], []
+         ,[{'call',4,{'remote',4,{'atom',4,SupMod},{'atom',4,'start_link'}},[]}]
+         }
+        ]
+       }
+      ,{'function',5,'stop',1,[{'clause',5,[{'var',5,'_'}],[],[{'atom',5,'ok'}]}]}
+      ,{'eof',6}
```
Comment (jamesaimonetti) [inline kazoo_appex/src/appex_bluepak.erl]:
```erlang
  appex_bluepak.erl:390: The call compile:forms
         ([{'eof', 6} |
           {'attribute',
            1 | 2 | 3,
            'behaviour' | 'export' | 'file' | 'module',
            atom() |
            [{'start', 2} | {'stop', 1}, ...] |
            {[[any()] | char()], 1}} |
           {'function',
            4 | 5,
            'start' | 'stop',
            1 | 2,
            [{'clause',
              4 | 5,
              [{'var', 4 | 5, '_'}, ...],
              [],
              [{'atom', 5, 'ok'} |
               {'call', 4,
                {'remote', 4,
                 {'atom', 4, atom()},
                 {'atom', 4, 'start_link'}},
                []},
               ...]},
             ...]},
           ...]) does not have a term of type 
          [{'attribute', erl_anno:anno(), atom(), _} |
           {'function',
            erl_anno:anno(),
            atom(),
            byte(),
            [{'clause',
              erl_anno:anno(),
              [{_, _} | {_, _, _} | {_, _, _, _} | {_, _, _, _, _}],
              [[{_, _} | {_, _, _} | {_, _, _, _} | {_, _, _, _, _},
                ...]],
              [{'nil', erl_anno:anno()} |
               {atom(), erl_anno:anno(), _} |
               {'bc' | 'call' | 'case' | 'cons' | 'lc' | 'map' |
                'match' | 'named_fun' | 'op' | 'record' | 'record_index',
                erl_anno:anno(),
                _, _} |
               {'op', erl_anno:anno(), atom(), _, _} |
               {'receive',
                erl_anno:anno(),
                [{'clause', erl_anno:anno(), _, _, _}, ...],
                _,
                [any(), ...]} |
               {'record', erl_anno:anno(), _, atom(), [any()]} |
               {'record_field', erl_anno:anno(), _, atom(), {_, _, _}} |
               {'try',
                erl_anno:anno(),
                [any(), ...],
                [{'clause', erl_anno:anno(), _, _, _}],
                [{'clause', erl_anno:anno(), _, _, _}],
                [any()]},
               ...]},
             ...]}] (with opaque subterms) as 1st argument
```

I think 'eof' is part of the `erl_parse:form_info()` type and not expected as an argument to `compile:forms/1`? Is this tuple needed?

File: kazoo_appex/src/appex_client_app_loader.erl [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+%% @end
+%%------------------------------------------------------------------------------
+-spec stop(any()) -> any().
+stop({?MODULE, {App, AppMod}, State}) ->
+    appex_app_master:stop_app(App, {AppMod, State}).
+
+try_fetch_license(App) ->
+    case appex_client_client:fetch_license(App) of
+        {'ok', RespBody} ->
+            JObj = kz_json:decode(RespBody),
```
Comment (jamesaimonetti) [inline kazoo_appex/src/appex_client_app_loader.erl]: I think fetch_license returns JObj already, no?

File: kazoo_appex/src/appex_bluepak.erl (hunk 1 above)
Comment (icehess) [reply kazoo_appex/src/appex_bluepak.erl]: Curious how did you get that? using dialyze-hard?

File: kazoo_stdlib/src/kz_term.erl [hunk 3]
```diff
@@ -425,6 +427,95 @@ is_true("true") -> 'true';
 is_true('true') -> 'true';
 is_true(_) -> 'false'.
 
+-spec from_centos_version(Version::ne_binary()) ->
```
Comment (jamesaimonetti) [inline kazoo_stdlib/src/kz_term.erl:430]: Quibble, but I see `kz_term` as Erlang term manipulation.

We have `kz_os` for running commands in the underlying operating system; maybe put this function there?

**Pull request merged**